.github/workflows/build.yml

```diff
@@ -78,12 +81,32 @@ jobs:
       - name: Test
         run: npm run test:ci
 
+      - name: Check git diff of Wing CLI
+        id: git-diff-wing-cli
+        run: |
+          if git diff --quiet ${{ github.event.before }} ${{ github.sha }} -- apps/wing; then
+            echo "::set-output name=diff::false"
+          else
+            echo "::set-output name=diff::true"
+          fi
+
       - name: Upload Wing CLI
         uses: actions/upload-artifact@v3
         with:
           name: wing
           path: apps/wing/*.tgz
 
+      - name: Generate the fig spec
+        if: ${{ steps.git-diff-wing-cli.outputs.diff == 'true' }}
+        run: cd apps/wing && node dist/cli.js generate-fig-spec >> new-fig-cli-generated.ts
```

Comment:
idea: What if this process runs in a separate workflow with a release trigger? Then it can just use the cli directly from npm to generate this spec.

Reply:
Fyi: once https://github.com/winglang/wing/pull/2755 is successfully merged I will apply the same technique with `nrwl/nx-set-shas@v3` action here